getdokan/dokan
# Feature: Withdraw charges

### All Submissions:






### Changes proposed in this Pull Request:

New feature withdraw charge.

* Closes https://github.com/weDevsOfficial/dokan-pro/issues/2481

* Dokan Pro PR: https://github.com/getdokan/dokan-pro/pull/2692

Video Links: [Introduction](https://www.awesomescreenshot.com/video/19641854?key=4f8c3161a585e45686f6e932f5daf16a), [Feature video](https://www.awesomescreenshot.com/video/19642008?key=22384fd5e0e484ec3075a931bb2bb2ef)

### How to test the changes in this Pull Request:

1. First do not set charge and try to make  a withdraw request. the experience will be as before.
2. Set withdraw charges and try to make request.
3. Check all the amount are alright as expected when applied charge.



### Changelog entry

> Admin can set a withdrawal charge (flat rate and percentage) for vendors so that admin can charge vendors for withdrawals based on the payment gateway and it will also reflect in the details report. Vendor can see how many charge will apply when request withdraw and show the charge and receivable amount in vendor dashboard list.


### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.


### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.
